low-level/sensors/impl/generic/node_hw.py

```diff
@@ -1174,36 +1176,33 @@ def _parse_disk_info(self, index, date, _time, sensor, sensor_num, event, status
         """Parse out Disk related changes that gets reaflected in the ipmi sel list"""
 
         sensor_id = self.sensor_id_map[self.TYPE_DISK][sensor_num]
+        disk_slot = re.search(r'/d+', sensor_id)
+        if disk_slot:
+            disk_slot = disk_slot.group()
+        disk_name = sensor_id.replace('Status', f'({sensor_num})')
 
         common, specific, specific_dynamic = self._get_sensor_props(sensor_id)
         if common:
-            disk_sensors_list = self._get_sensor_list_by_entity(common['Entity ID'])
-            disk_sensors_list.remove(sensor_id)
-
-            description = f"{event} - {status}"
-
             if not specific:
                 specific = {"States Asserted": "N/A", "Sensor Type (Discrete)": "N/A"}
             specific_info = specific
-            alert_severity_dict = {
-                ("Drive Present", "Asserted"): ("insertion", "informational"),
-                ("Drive Present", "Deasserted"): ("missing", "critical"),
-                }
             resource_type = NodeDataMsgHandler.IPMI_RESOURCE_TYPE_DISK
-            fru = self.ipmi_client.is_fru(self.fru_map[self.TYPE_DISK])
+            try:
+                alert = alert_for_event[self.TYPE_DISK][event][status]
+            except KeyError:
+                alert = Alert("fault", "informational", f"{event} - {status}",
+                              "unknown", DEFAULT_RECOMMENDATION)
+            except Exception as e:
+                logger.exception(e)
             info = {
                 "resource_type": resource_type,
-                "fru": fru,
+                "fru": self.ipmi_client.is_fru(self.fru_map[self.TYPE_DISK]),
                 "resource_id": sensor,
                 "event_time": self._get_epoch_time_from_date_and_time(date, _time),
-                "description": description
+                "description": alert.description.format(disk_slot, disk_name),
+                "impact": alert.impact.format(disk_name),
+                "recommendation": alert.recommendation
             }
-            if (event, status) in alert_severity_dict:
```

Comment:
where this code is covered? line 1201-1206

Reply:
in the `node_hw_alerts_info.py` file that I added.

Reply:
understood